src/lib/substrate/context/types.ts

```diff
@@ -0,0 +1,100 @@
+import { ApiPromise } from '@polkadot/api';
+import { InjectedAccountWithMeta, InjectedExtension } from '@polkadot/extension-inject/types';
+import { KeyringPair, KeyringPair$Meta } from '@polkadot/keyring/types';
+import type { DefinitionRpcExt } from '@polkadot/types/types';
+import { Keyring } from '@polkadot/ui-keyring/Keyring';
+
+enum KeyringStatus {
+  Idle = 'IDLE',
+  Loading = 'LOADING',
+  Ready = 'READY',
+  Error = 'ERROR'
+}
+
+enum ApiStatus {
+  Idle = 'IDLE',
+  ConnectInit = 'CONNECT_INIT',
+  Connecting = 'CONNECTING',
+  Ready = 'READY',
+  Error = 'ERROR',
+  Disconnected = 'DISCONNECTED'
+}
+
+enum ActionType {
+  ConnectInit = 'CONNECT_INIT',
+  Connect = 'CONNECT',
+  ConnectSuccess = 'CONNECT_SUCCESS',
+  ConnectError = 'CONNECT_ERROR',
+  ConnectFail = 'CONNECT_FAIL',
+  SetKeyringLoading = 'SET_KEYRING_LOADING',
+  SetKeyringReady = 'SET_KEYRING_READY',
+  SetKeyringError = 'SET_KEYRING_ERROR',
+  SetSelectedAccount = 'SET_SELECTED_ACCOUNT',
+  SetAccounts = 'SET_ACCOUNTS',
+  SetExtensions = 'SET_EXTENSIONS'
+}
+
+interface APIError extends Error {
+  target: {
+    url: string;
+  };
+}
+
+interface KeyringPairMeta extends KeyringPair$Meta {
+  name: string;
+}
+
+interface _KeyringPair extends KeyringPair {
+  meta: KeyringPairMeta;
+}
+
+type Action =
```

Comment:
NIT: could you just add some spacing between the types?

Reply:
Done!